### ISSUE TITLE
Regression: 3.5.0-RC1 translations sometimes missing

### ISSUE THREAD

**Issue by Rytiss** [cakephp/cakephp#11026]
This is a (multiple allowed):

* [x] bug
* [ ] enhancement
* [ ] feature-discussion (RFC)

* CakePHP Version: 3.5.0-RC1
* Platform and Target: PHP 5.6.31, Apache

### What you did
I set locale to Lithuanian (either I18n::setLocale('lt') or I18n::locale('lt') - doesn't matter which option). The following code only displays some of the translations in 3.5.0-RC-1, but always works in 3.4.12:

```
$conditions = ['parent_id IS NULL'];
$this->paginate = [
  'finder' => 'translations',
  'sortWhitelist' => ['Categories_name_translation.content'],
  'conditions' => $conditions,
  'order' => ['Categories_name_translation.content'],
];
$this->set('menuItems', $this->paginate($this->Categories));
```

View code:
```
<?php foreach ($menuItems as $menu): ?>
<?= $menu->translation('lt')->name ?>
<?php endforeach; ?>
```

Also tried `<?= $menu->name ?>` for rendering.

### What happened
In 3.4.12 view renders all translations correctly:
Ašies montavimas/vairavimo mechanizmas/ratai
Išmetimo sistema
PTO
Suspausto oro sistema
Techninės priežiūros dalys
aušinimo sistema
ašies pavara
degalų tiekimo sistema
diržinė pavara
dviratė transporto priemonės
elektrinė pavara
elektros įranga

In 3.5.0-RC1 view renders only some of the translations and the others are blank:

Ašies montavimas/vairavimo mechanizmas/ratai
(blank)
Išmetimo sistema
(blank)
(blank)
Techninės priežiūros dalys
aušinimo sistema
(blank)
degalų tiekimo sistema
diržinė pavara
(blank)
(blank)
elektros įranga

### What you expected to happen
All translations should be rendered.

**Comment by dereuromark** [comment]
This has been fixed in https://github.com/cakephp/cakephp/releases/tag/3.4.12 which was released after 3.5 RC, thus the missing fix. the next RC will sure contain the fix then, as well.

**Comment by ADmad** [comment]
Closing as issue has already been fixed and will be available in next release.
